papis/commands/doctor.py

```diff
@@ -0,0 +1,374 @@
+    """
+    Check for duplicated keys in `doctor-duplicated-keys-check`
+    """
+    keys = papis.config.getlist("doctor-duplicated-keys-keys")
+    folder = doc.get_main_folder()
+    results = []  # type: List[Error]
+    for key in keys:
+        if doc[key] in DUPLICATED_KEYS_SEEN[key]:
+            results.append(Error(name=DUPLICATED_KEYS_NAME,
+                                 msg=("Key '{}' is duplicated ({})."
+                                      .format(key, doc[key])),
+                                 suggestion_cmd=("papis edit {}:'{}'"
+                                                 .format(key, doc[key])),
+                                 payload=key,
+                                 fix_action=lambda: None,
+                                 path=folder or "",
+                                 doc=doc))
+        else:
+            DUPLICATED_KEYS_SEEN[key].append(str(doc[key]))
+    return results
+
+
+BIBTEX_TYPE_CHECK_NAME = "bibtex-type"
+
+
+def bibtex_type_check(doc: papis.document.Document) -> List[Error]:
+    """
+    Check that the types are compatible with bibtex or biblatex
+    type descriptors.
+    """
+    types = papis.bibtex.bibtex_types
+    folder = doc.get_main_folder()
+    results = []
+    if doc["type"] not in types:
+        results.append(Error(name=BIBTEX_TYPE_CHECK_NAME,
+                             path=folder or "",
+                             msg=("Document type '{}' is not"
+                                  " a valid bibtex type"
+                                  .format(doc["type"])),
+                             suggestion_cmd=("papis edit --doc-folder {}"
+                                             .format(folder)),
+                             fix_action=lambda: None,
+                             payload=doc["type"],
+                             doc=doc))
+    return results
+
+
+HTML_CODE_REGEX = re.compile(r"&[a-z_A-Z0-9]+;")
+HTML_CODES_CHECK_NAME = "html-codes"
+
+
+def html_codes_check(doc: papis.document.Document) -> List[Error]:
+    """
+    Checks that the keys in "doctor-html-code-keys" do not contain
+    any html codes like &amp; etc.
+    """
+    results = []
+    folder = doc.get_main_folder()
+
+    def _fix(key: str) -> Callable[[], None]:
+
+        def __fix() -> None:
+            db = papis.database.get()
+            val = html.unescape(doc[key])
+            logger.info("Setting '%s' to '%s'", key, val)
+            doc[key] = val
+            doc.save()
+            db.update(doc)
+
+        return __fix
+
+    for key in papis.config.getlist("doctor-html-codes-keys"):
+        if doc[key]:
+            m = HTML_CODE_REGEX.findall(str(doc[key]))
+            if m:
+                results.append(Error(name=HTML_CODES_CHECK_NAME,
+                                     path=folder or "",
+                                     msg=("Field '{}' contains html codes '{}'"
+                                          .format(key, m)),
```

Comment:
```suggestion
                                     msg=("Field '{}' contains html codes {}"
                                          .format(key, m)),
```
? `m` should already be a list of quoted strings, so the extra quote may look weird..

Reply:
agreed!